dockstore-client/src/main/java/io/dockstore/client/cli/nested/CromwellLauncher.java

```diff
@@ -0,0 +1,234 @@
+        super(abstractEntryClient, language, script);
+    }
+
+    /**
+     * Creates a local copy of the Cromwell JAR (May have to download from the GitHub).
+     * Uses the default version unless a version is specified in the Dockstore config.
+     */
+    @Override
+    public void initialize() {
+        // initialize cromwell location from ~/.dockstore/config
+        INIConfiguration config = Utilities.parseConfig(abstractEntryClient.getConfigFile());
+        String cromwellVersion = config.getString("cromwell-version", DEFAULT_CROMWELL_VERSION);
+        String cromwellLocation =
+                "https://github.com/broadinstitute/cromwell/releases/download/" + cromwellVersion + "/cromwell-" + cromwellVersion + ".jar";
+        if (!Objects.equals(DEFAULT_CROMWELL_VERSION, cromwellVersion)) {
+            System.out.println("Running with Cromwell " + cromwellVersion + " , Dockstore tests with " + DEFAULT_CROMWELL_VERSION);
+        }
+
+        // grab the cromwell jar if needed
+        String libraryLocation =
+                System.getProperty("user.home") + File.separator + ".dockstore" + File.separator + "libraries" + File.separator;
+        URL cromwellURL;
+        String cromwellFileName;
+        try {
+            cromwellURL = new URL(cromwellLocation);
+            cromwellFileName = new File(cromwellURL.toURI().getPath()).getName();
+        } catch (MalformedURLException | URISyntaxException e) {
+            throw new RuntimeException("Could not create " + LAUNCHER_NAME + " location", e);
+        }
+        String cromwellTarget = libraryLocation + cromwellFileName;
+        File cromwellTargetFile = new File(cromwellTarget);
+        if (!cromwellTargetFile.exists()) {
+            try {
+                FileUtils.copyURLToFile(cromwellURL, cromwellTargetFile);
+            } catch (IOException e) {
+                throw new RuntimeException("Could not download " + LAUNCHER_NAME + " location", e);
+            }
+        }
+        exectionFile = cromwellTargetFile;
+    }
+
+    @Override
+    public String buildRunCommand() {
+        final List<String> runCommand;
+        // Don't use imports option for WDL, only for CWL
+        if (importsZip == null || abstractEntryClient instanceof ToolClient || Objects.equals(languageType, LanguageType.WDL)) {
+            runCommand = Lists.newArrayList(primaryDescriptor.getAbsolutePath(), "--inputs", provisionedParameterFile.getAbsolutePath());
+        } else {
+            runCommand = Lists.newArrayList(primaryDescriptor.getAbsolutePath(), "--inputs", provisionedParameterFile.getAbsolutePath(), "--imports", importsZip.getAbsolutePath());
+        }
+        // run a workflow
+        System.out.println("Calling out to " + LAUNCHER_NAME + " to run your workflow");
+
+        // Currently Cromwell does not support HTTP(S) imports
+        // https://github.com/broadinstitute/cromwell/issues/1528
```

Comment:
This comment seems out of place -- I don't see any code that handles the http imports issue.

Reply:
Also, the issue is closed (fixed).

Reply:
Some of the code will show as green because I moved it, even though I didn't actually change that portion of the code. I'll still remove this as it does seem out of place.